js/test/Ice/idleTimeout/Client.ts

```diff
@@ -0,0 +1,137 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+import { Ice } from "ice";
+import { Test } from "./Test.js";
+import { TestHelper } from "../../Common/TestHelper.js";
+
+const test = TestHelper.test;
+
+// We verify that the idle check aborts the connection when the connection (here client connection) remains idle for
+// longer than idle timeout. Here, the client has an idle timeout of 1s and idle checks enabled. We intentionally
+// misconfigure the server with an idle timeout of 3s to send heartbeats every 1.5s, which is too long to prevent
+// the client from aborting the connection.
+async function testConnectionAbortedByIdleCheck(
+    proxyString: string,
+    properties: Ice.Properties,
+    helper: TestHelper,
+): Promise<void> {
+    const output = helper.getWriter();
+    output.write("testing that the idle check aborts a connection that does not receive anything for 1s... ");
+
+    let completedResolve: () => void;
+    const completedPromise = new Promise<void>((resolve) => (completedResolve = resolve));
+
+    const TestI = class extends Test.TestIntf {
+        async sleep(msecs: number, current?: Ice.Current): Promise<void> {
+            setTimeout(() => {
+                completedResolve();
+            }, msecs);
+            await completedPromise;
+        }
+
+        shutdown(current: Ice.Current): void | PromiseLike<void> {
+            current.adapter.getCommunicator().shutdown();
```

Comment:
It would be cleaner to define a separate interface with just `sleep`.

This implementation is not supposed to received shutdown or setCallback calls.